### ISSUE TITLE
release 2020-02 follow-up

### ISSUE THREAD

**Issue by stonebig** [winpython/winpython#819]
release date: Q2 2020 (may 23rd, 2020)

Expected:
- Jupyterlab-2.1, Jedi-0.17.0+
- Python-3.8 reaches parity of packages with Python-3.7 :
   - [x] Tensorflow-cpu-2.2.0 
   - [x] Cartopy-0.18.0 
   - [x] rpy2-3.4.0.dev0  (tentative https://github.com/rpy2/rpy2/pull/685)
- Pytorch-1.5
- better Jupyter Notebook experience in Python-VSCode (ipywidgets support, but you must have opened & ran a first Python file, before opening a notebook)

Cleanup:
- maybe a cleanup based on factual numbers https://pyviz.org/tools.html
- last Python-3.7 release

Focus of the release:
- cleanup & simplify further the build steps: 
   - [x] packages.ini update should be optional for users, 
   - [x] packages.ini update should be automated in build process
- fix default missing DLL https://github.com/winpython/winpython/issues/828
- impact of coming pip "stricter rules" change

Exploratory work:
- flask eco-system
- jedi & python-langage-server ... jedi-0.17.0 in the release

**Comment by stonebig** [comment]
audit-trail:

Exploration
- modin
- ray

**Comment by stonebig** [comment]
build1 (2020-04-xx, Pytorch on Python-3.8, Matplotlib-3.2rc3)

Changes from WinPython 2020-01:
- Infrastructure: 
  - on wppm, packages labels when not present  in internal databases packages.ini: 
      . are taken in  Summary information of the installed package
      . otherwise taken live from pypi summary (to be checked on your situation, not super fast, count 15 seconds)
![image](https://user-images.githubusercontent.com/4312421/78698010-f58fed80-7901-11ea-873b-f795b91ff1bb.png)

- New Packages: 
   - on Python-3.8 : pyarrow, feather_format, torch
   - pydeck, flask_restx
- Upgraded Packages: 
   - Matplotlib-3.2.0rc3, numpy-1.18.1, pandas-1.0.1, ipython-7.12.0, Jupyterlab-1.2.6
   - seaborn-0.10.0, statsmodels-0.11.0, dask-2.10.1, datashader-0.10.0, panel-0.8.0,
   - cython-0.29.15, numba-0.48.0
   - PyQt5-5.14.1 , PyQtgraph-0.11.0rc0
- variation per version, for build1:
  -  Spyder-3 is in PyQt5 only, (and Spyder-4 a problem till october 2020)
  
Areas of particular interest for testers:
- matplotlib-3.2rc3
- pandas-1, Qt-5.14.1

Next build effort:
- python-3.8.2, 
- python-vscode with the February Data-Science (Jupyter) improvements
- Bokeh-2, Matplotlib-3.2 final

**Comment by stonebig** [comment]
build1b : (2020-04-10)
- correct error of build on 32 bit versions

**Comment by stonebig** [comment]
build2 (2020-04-1x)

Changes from WinPython 2020-01 build1:
- Infrastructure: 
  - fix the user settings directory for Python-3.8.2 and Python-3.9 (a change in cpython)
- New Packages: 
   - on Python-3.8 : geoviews + cartopy
- Upgraded Packages: 
- variation per version:
  -  Tensorflow and Pytorch in standard branch only,
  -  portable VSCode in "cod" branch only. (but you can copy the "\t\cod" directory from one to the other)

Hope for Rc:
- maybe rpy2 comeback is not impossible.
- Pandas-1.0.4, Pytorch-1.5, Tensorflow-2.2

**Comment by stonebig** [comment]
interlude: python-3.9.0a6 (2020-05-01)

This version brings the real new things for Python-3.9:

- PEP 584 https://www.python.org/dev/peps/pep-0584/,
Union Operators in dict
- PEP 585 https://www.python.org/dev/peps/pep-0585/,
Type Hinting Generics In Standard Collections
- PEP 593 https://www.python.org/dev/peps/pep-0593/,
Flexible function and variable annotations
- PEP 616 https://www.python.org/dev/peps/pep-0616/,
String methods to remove prefixes and suffixes
- PEP 617 https://www.python.org/dev/peps/pep-0617/,
New PEG parser for CPython
- sqlite-3.31.1 with json module

 MD5                             | SHA-1                                    | SHA-256                                                          | Binary                          | Size              
---------------------------------|------------------------------------------|------------------------------------------------------------------|---------------------------------|-------------------
a9e49d369f6fa2ab9776ab2799b16f00 | 3bbaa5b0951f896dd5473552adcf362c33e9d7ec | c7d92acb198e6b72ed465c1cb20558165cfff4d483f856c3d28fcf3d04423e6f | Winpython64-3.9.0.0dota6.exe    |   24 428 549 Bytes
e741be6ba7c5cdcffbdb409dd97419f4 | 44e0f07149b72a1b5999f5854880c8eea5032bf6 | 94b5705435b0834f0b5dfc8e134f09d103e1e92b48e529baa1f555355f168b90 | Winpython32-3.9.0.0dota6.exe    |   23 418 106 Bytes

**Comment by stonebig** [comment]
build3 2020-05-01 

 MD5                             | SHA-1                                    | SHA-256                                                          | Binary                          | Size              
---------------------------------|------------------------------------------|------------------------------------------------------------------|---------------------------------|-------------------
3d3e8d13171214ee136397d556d7fbf5 | 68a800d996e421ccf82fe33aa878d14340612d5c | 4f05f4e9965781edd1681ad014873d313329b47af1d0cb01f28f37592221110e | Winpython64-3.8.3.0dotb3.exe    |   24 135 337 Bytes
fc4db365e914c4424928ef08df5b4c57 | 6d27cb6bb5837c1544dc853e53d090a7889f79eb | ba64ebbd17197a6b6f7f15ce07050733bdb6de37f1667ed718ffdb44eb989572 | Winpython32-3.8.3.0dotb3.exe    |   23 104 170 Bytes

**Comment by stonebig** [comment]
You can see python-vscode feature here : https://www.youtube.com/watch?v=6Pvrk1Qt-B4&feature=youtu.be&ocid=AID3006292

**Comment by stonebig** [comment]
rc (2020-05-118/20) 


Changes from WinPython 2020-01 build3:
- Infrastructure: 
  - python-3.8.3 final
- New Packages: 
- Upgraded Packages: 
  - python-vscode-2020.5.78807 (bug... now requires a .py file to be open before a .ipynb is open and ipywidgets work)
  - tensorflow-cpu-2.2.0 final, Cartopy-0.18.0 final
  - scikit_learn-0.23.0, scikit_image-0.17.2
- variation per version:
  -  Tensorflow and Pytorch not in "cod" branch 
  -  portable VSCode in "cod" branch only. (but you can copy the "\t\cod" directory from one to the other)

Interesting for users:
- Python-3.8.3,
- do not complaint on Spyder team, as PyQt-5.14.2 and  Jedi-0.17.0 and Python-Language-server-0.32.0 are not supported.
- "cod" version may be delayed a few days to wait for hypothetical bug fix

**Comment by hiccup7** [comment]
@stonebig Did the RC build crash?  I only see the "dot" versions.

**Comment by stonebig** [comment]
Hi @hiccup7 ,   it's ready under my foot since 5 days as I was hoping for a jedi-0.17.1, a pandas-1.04, and a python-vscode that doesn't want to be fixed.

**Comment by hiccup7** [comment]
Hi @stonebig  I found the remaining RC build binaries on SourceForge.  Please post the MD5-SHA hashes for them.  Thanks!

**Comment by stonebig** [comment]
In there ? https://github.com/winpython/winpython/issues/819#issuecomment-629770375

**Comment by hiccup7** [comment]
Yes, they are there.  Sorry to bother you.

**Comment by stonebig** [comment]
2020-05-30 WinPython 2020-02 release is equal to rc

As there has been no major complaint, the rc binaries are decelared ok and will be renamed as such in a few minutes

**Comment by hannahzacharski** [comment]
> release date: Q2 2020 (may 23rd, 2020)
> 
> Expected:
> 
> * Jupyterlab-2.1, Jedi-0.17.0+
> * Python-3.8 reaches parity of packages with Python-3.7 :
>   
>   * [x]  Tensorflow-cpu-2.2.0
>   * [x]  Cartopy-0.18.0
>   * [x]  rpy2-3.4.0.dev0  (tentative [rpy2/rpy2#685](https://github.com/rpy2/rpy2/pull/685))
> * Pytorch-1.5
> * better Jupyter Notebook experience in Python-VSCode (ipywidgets support, but you must have opened & ran a first Python file, before opening a notebook)
> 
> Cleanup:
> 
> * maybe a cleanup based on factual numbers https://pyviz.org/tools.html
> * last Python-3.7 release
> 
> Focus of the release:
> 
> * cleanup & simplify further the build steps:
>   
>   * [x]  packages.ini update should be optional for users,
>   * [x]  packages.ini update should be automated in build process
> * fix default missing DLL #828
> * impact of coming pip "stricter rules" change
> 
> Exploratory work:
> 
> * flask eco-system
> * jedi & python-langage-server ... jedi-0.17.0 in the release

**Comment by RoyiAvital** [comment]
In the packages of [`WinPython 3.7.7.1`](https://github.com/winpython/winpython/blob/master/changelogs/WinPython-64bit-3.7.7.1.md) I can see it says `numpy | 1.18.4+mkl`.

Yet when I do:

```
np.show_config()
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    library_dirs = ['C:\\projects\\numpy-wheels\\numpy\\build\\openblas']
    libraries = ['openblas']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
blas_opt_info:
    library_dirs = ['C:\\projects\\numpy-wheels\\numpy\\build\\openblas']
    libraries = ['openblas']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    library_dirs = ['C:\\projects\\numpy-wheels\\numpy\\build\\openblas']
    libraries = ['openblas']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
lapack_opt_info:
    library_dirs = ['C:\\projects\\numpy-wheels\\numpy\\build\\openblas']
    libraries = ['openblas']
    language = f77
    define_macros = [('HAVE_CBLAS', None)]
```

Also:

```
np.__version__
Out[3]: '1.16.6'
```

And:

```
pip show numpy
Name: numpy
Version: 1.16.6
Summary: NumPy is the fundamental package for array computing with Python.
Home-page: https://www.numpy.org
Author: Travis E. Oliphant et al.
Author-email: None
License: BSD
Location: d:\applications\programming\winpython\wpy64-3771\python-3.7.7.amd64\lib\site-packages
```

Any idea?

**Comment by stonebig** [comment]
The changelog is made from "pip list", what does it say now on your installation ?

**Comment by stonebig** [comment]
I would suspect:
- either you had an update on your installation,
- either Torch or Tensorflow do some sort of 'vendored' thing.

![image](https://user-images.githubusercontent.com/4312421/85942609-9d79ed80-b92a-11ea-9911-44452fbb3a14.png)

**Comment by RoyiAvital** [comment]
I only installed OpenCV which has a requirement on numpy which is `>=`.
What about MKL? Does it show on your system?

**Comment by hiccup7** [comment]
In my WinPython 3.7.7.1, np.__version__ produces "1.18.4", and np.show_config() shows MKL.

**Comment by RoyiAvital** [comment]
Strange... All I installed was `OpenCV` and `DLib`.
